### PR TITLE
action-docs 2.5.0 (new formula)

### LINKED ISSUE #666: Homebrew formula for action-docs
Please add homebrew support!

```
brew install action-docs
```

### PR DESCRIPTION
**Pull request by nitrocode**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

- Closes https://github.com/npalm/action-docs/issues/666#issuecomment-2435928689

### CODE REVIEW

File: Formula/a/action-docs.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  def install
+    system "npm", "install", *std_npm_args
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    assert_predicate bin/"action-docs", :exist?
+    assert_predicate bin/"action-docs", :executable?
+
```
Comment (chenrui333) [inline Formula/a/action-docs.rb:18]:
```suggestion
```

File: Formula/a/action-docs.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      runs:
+        using: "composite"
+        steps:
+          - id: random-number-generator
+            run: echo "random-id=$(echo $RANDOM)" >> $GITHUB_OUTPUT
+            shell: bash
+    EOT
+
+    output = shell_output("#{bin}/action-docs 2>&1")
+    assert_match(/Example input description/, output)
```
Comment (chenrui333) [inline Formula/a/action-docs.rb:37]:
```suggestion
    assert_match "Example input description", shell_output("#{bin}/action-docs")
```

File: Formula/a/action-docs.rb (hunk 1 above)
Comment (nitrocode) [reply Formula/a/action-docs.rb]: Thank you. Addressed.

File: Formula/a/action-docs.rb (hunk 2 above)
Comment (nitrocode) [reply Formula/a/action-docs.rb]: Thank you. Addressed.

**Comment by chenrui333** [comment]
```
  action-docs
    * line 33, col 60: Use `bin/"action-docs"` instead of `"#{bin}/action-docs"`
```

**Comment by nitrocode** [comment]
Thanks @chenrui333. PR checks are all green now. Please re-review when you get a chance.

**Review by chenrui333 — APPROVED** [review]

File: Formula/a/action-docs.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  depends_on "node"
+
+  def install
+    system "npm", "install", *std_npm_args
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    test_file = testpath/"action.yml"
+    test_file.write <<~EOT
```
Comment (chenrui333) [inline Formula/a/action-docs.rb:17]:
```suggestion
    test_file.write <<~YAML
```

**Comment by nitrocode** [comment]
@chenrui333 thank you for the review. I see that you approved and then force pushed which then removed the approval. Would you mind reviewing again please?

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
we dont need commit fresh, I was just making some test code style changes, but it should be all good. thanks @nitrocode.

**Comment by nitrocode** [comment]
Thanks!

**Pull request merged**